### PR TITLE
[v2] responsive image types

### LINKED ISSUE #3740: Convert responsive image type names to "fixed" and "fluid"
The current "resolutions" and "sizes" are derived loosely from parts of the responsive image specs but do a mostly poor job of indicating to people what to use them for. Let's fix this before it's too late.

### PR DESCRIPTION
**Pull request by m-allanson**
~~Work in progress.~~ Ready for review

Todo:

- [x] Rename `resolutions` to `fixed`
- [x] Rename `sizes` to `fluid`
- [x] rename responsiveResolutions to resolutions (deprecation)
- [x] rename responsiveSizes to sizes (deprecation)
- [x] Tests pass
- [x] update `image-processing` example to Gatsby v2 (#5589)
- [x] update `using-contentful` example to Gatsby v2 (#5606)
- [x] `image-processing` example works
- [x] `using-contentful` example works
- [x] `using-gatsby-image` example works
- [x] `gatsbygram` works

Closes #3740

### CODE REVIEW

**Comment by pieh** [comment]
Just quick note - renaming responsiveResolutions to resolutions can cause problems, because currently depracated `responsiveResolutions` is not exactly like `resolutions` field (for example it's missing `srcWebp` field). My idea was to to move `fixed` field definition to dedicated variabls and it in both `fixed` in `resolutions` fields (and just adding deprecation notice in `resolutions` field)

**Comment by axe312ger** [comment]
Sweet, this lowers the learning barrier a lot. The new names are way better 👍 

Since this touches kinda the same parts, can we add an extra value to gatsby-image to replace the base64 property of resolutions/sizes?

This would be helpful for sqip, since at the moment the following is needed:
```
<Img
  resolutions={{
    ...image.resolutions,
    base64: image.sqip.dataURI
  }}
/>
```



```
<Img resolutions={image.resolutions} preview={image.sqip.dataURI} />
```

I guess it would need to go to a seperate PR? Or squeeze it in here?

**Comment by KyleAMathews** [comment]
>can we add an extra value to gatsby-image to replace the base64 property of resolutions/sizes?

It seems optimizing for the common case is what we want?

Or are you saying that the `preview` prop would override `base64` if set?

**Comment by axe312ger** [comment]
@KyleAMathews correct. That way everybody can overwrite the base64 as they want. Maybe call it `dataURI` since it has to be one, no matter which kind.

**Comment by KyleAMathews** [comment]
>That way everybody can overwrite the base64 as they want. Maybe call it dataURI since it has to be one, no matter which kind.

I'm not a huge fan of adding special-case APIs when a bit of code can work around the use case. I think people should be fine with the spread override solution. It's roughly the same amount of code & allows us to avoid adding an extra prop.

**Comment by m-allanson** [comment]
Ok, this is ready for review. It's a fairly chunky PR, the main work is in the two `extend-node-types.js` files.

As part of this I've [renamed the ImageSharp and GatsbyContentful fragments](https://github.com/gatsbyjs/gatsby/pull/5250/files#diff-e0b11defb77b24c6fd2e878c0f6c3fc0), I guess this should be added to the list of breaking changes? Or do we want to keep both the old and new fragments?

**Comment by axe312ger** [comment]
I vote for keeping the new fragments, only. The new names are more descriptive :)

**Comment by m-allanson** [comment]
> I vote for keeping the new fragments, only. The new names are more descriptive :)

Sounds good to me :) It looks like that's the previous fragment name change was handled too.

> Did you check if you replaced all documentation, even of untouched plugins which may just use the fragments in their demos etc.?

Yeah I did a lot of find-and-replace across the whole repo, so should have picked up examples, plugins and blog posts that used the old names.

**Comment by KyleAMathews** [comment]
Yeah — might as well do a clean break — it's a straightforward upgrade — find/replace "resolutions" for "fixed" and "sizes" for "fluid".

File: docs/blog/2017-02-21-1-0-progress-update-where-came-from-where-going/index.bkup [hunk 1]
```diff
@@ -1,139 +0,0 @@
----
-title: "Gatsby 1.0 update"
-date: "2017-02-23"
-author: "Kyle Mathews"
----
-
-*hint — the solution isn't reverting to an idealized "simpler" way of
```
Comment (KyleAMathews) [inline docs/blog/2017-02-21-1-0-progress-update-where-came-from-where-going/index.bkup]: This is still a great article... maybe I should polish it up and publish it 🤔

**Comment by KyleAMathews** [comment]
@stefanoverna heads up on this breaking change — the old names, "resolutions" and "sizes" will still work but I think it'd be nice if gatsby-source-datocms also upgraded since the new names are much nicer.

**Comment by KyleAMathews** [comment]
Still work in gatsby-image that is.

File: docs/blog/2017-02-21-1-0-progress-update-where-came-from-where-going/index.bkup (hunk 1 above)
Comment (m-allanson) [reply docs/blog/2017-02-21-1-0-progress-update-where-came-from-where-going/index.bkup]: It just needs an updated title: _Gatsby ~1.0~ 2.0 update_ :)

**Comment by stefanoverna** [comment]
@KyleAMathews gatsby-source-datocms 1.1.9 also supports the new names!

**Comment by KyleAMathews** [comment]
@stefanoverna sweet!

**Pull request merged**